### DISCUSSION TITLE
Memory allocation problem - Flash_transparent_PNG.ino.elf section `.bss' is not within region `dram0_0_seg'

### DISCUSSION THREAD

**Discussion by edmond-vanas** [Bodmer/TFT_eSPI#2625] [Q&A - General]
Hi,

I'm trying to learn how to use color PNG's in my code. While going through the examples (panda.h, spongebob.h) from the TFT_eSPI library in the Arduino IDE, I got this error while compiling:

.../Library/Arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld: address 0x3fffd9c0 of /private/var/folders/mk/80q12ly173z561m6_dv90q6m0000gn/T/arduino/sketches/046004A340F477238E6DD8995CDB4746/Flash_transparent_PNG.ino.elf section `.bss' is not within region `dram0_0_seg'
collect2: error: ld returned 1 exit status

Seems to me the board I'm trying to compile this for is running out of memory to put the PNG in. Would that be correct?

I've tried to compile it on a WEMOS D1 Mini and a NODEMCU-ESP32, both with the same result. What am I doing wrong?

Thanks a lot!

Regards,

Edmond

**Comment by Bodmer** [comment]
The ESP8266 is marginal on having sufficient RAM for decoding a PNG file. The message indicates this.

The examples should work fine on an ESP32 but maybe you are using a very old version of the ESP32 board package?

**Reply by edmond-vanas** [reply]
My apologies. I forgot to reply to you. I think you're right. These are cheap boards, too.
